### ISSUE TITLE
Unexpected Drop in Accuracy During ResNet20 Training on CIFAR-10

### ISSUE THREAD

**Issue by TSommariva** [Adamdad/neumeta#4]
Hi,
I am trying to replicate the results using the code provided in the repository, but unfortunately, I am unable to achieve the expected results.
I am testing a ResNet20 on CIFAR-10 with an initial accuracy of 92.60%. However, using the script `neumeta/train_cifar10.py`,the generated models consistently have an accuracy between 75% and 76%, regardless of the `hidden_dim` value; even when it is set to 64, matching the original pretrained model. 
Additionally, even when I calculate accuracy on the training set, I am never able to achieve more than 81%.
These values represent the initial accuracies, the accuracy and loss graphs consistently oscillate around their starting values, suggesting that the hypernetwork is not learning properly.
I am unable to determine why this is happening. I am using the code cloned directly from GitHub, with only a few minor fix to make it work:

In `neumeta/neumeta/models/__init__.py`:

```python
fuse_module(model)
if path:
    if os.path.exists(path):
        print("Loading model from", path)
        state_dict = torch.load(path, map_location=torch.device('cpu'))
        load_checkpoint(model, state_dict)
```

When loading a model, this setup results in the model predicting the same class for all inputs. Moving `fuse_module()` after `load_checkpoint()` fixes this and correctly loads the network.

Additionally, these imports in `train_cifar10.py` cannot be resolved:
```
validate, validate_ensemble, sample_single_model, validate_merge
```

In `neumeta/utils/other_utils.py`, at line 41:
```python
config = OmegaConf.merge(config, cli_args)
```
The merge was not completing successfully, so I replaced it with:
```python
config = OmegaConf.merge(config, OmegaConf.create(cli_args))
```

Aside from these minor bug fixes, I have not modified anything else in the code, I hope I didn't introduce any bugs.

I am using the following configuration file (which, unless mistaken, should be the hyperparameters reported in the paper):

My_config.yaml:
```yaml
# Experiment settings
base_config: neumeta/config/base_config.yaml

# Model configurations
model:
  type: ResNet20
  pretrained_path: neumeta/pretrained_models/cifar10_resnet20-4118986f.pt   # Path to pre-trained model
  smooth: True

experiment:
  name: ${model.type}_
  num_epochs: 200
  log_interval: 200
  eval_interval: 1

# Training configurations
training:
  optimizer: adam
  learning_rate: 1e-3
  batch_size: 128
  coordinate_noise: 1.0
  #lr_steps: [100, 150]
  scheduler: cosine
  T_max: 200
  weight_decay: 1e-2
  clip_grad: 10.0
  save_model_path: experiments/${experiment.name} # Directory to save the trained model

# Hyper-model specifications (specific to NeRF)
hyper_model:
  type: resmlp
  input_dim: 6
  hidden_dim: 256
  num_layers: 5
  num_freqs: 16
  output_dim: 9
  ema_decay: 0.995  # Exponential moving average decay

# Experiment dimensions configuration
dimensions:
  range: [32, 64]  # Dimension range for the experiment
  test: 32  # Testing dimension
  norm: 64  # Normalization factor
  start: 64  # Starting dimension
```

base_config.yaml:
```yaml
# Experiment settings
experiment:
  name: ninr_resnet20_cifar10_32-64-4layer-5
  recononly: 0
  num_epochs: 30
  log_interval: 250
  eval_interval: 1
  seed: 42  # Random seed

# Model configurations
model:
  type: ResNet20
  pretrained_path: neumeta/pretrained_models/cifar10_resnet20-4118986f.pt  # Path to pre-trained model
  smooth: True

# Training configurations
training:
  learning_rate: 1e-3
  batch_size: 64
  coordinate_noise: 0.0
  #lr_steps: [100, 150]
  weight_decay: 1e-2
  clip_grad: 10.0
  save_model_path: neumeta/experiments/resnet20_cifar10_32-64-4layer-5 # Directory to save the trained model

# Hyper-model specifications (specific to NeRF)
hyper_model:
  input_dim: 6
  hidden_dim: 256
  num_layers: 4
  num_freqs: 16
  output_dim: 9
  ema_decay: 0.995  # Exponential moving average decay

  loss_weight:
    ce_weight: 1.0
    reg_weight: 0.01
    recon_weight: 1.0
    kd_weight: 0.1

# Experiment dimensions configuration
dimensions:
  range: [32, 64]  # Dimension range for the experiment
  test: 32  # Testing dimension
  norm: 64  # Normalization factor
  start: 64  # Starting dimension

```

If you could help me understand what I might be doing wrong and why the accuracy of the generated models are so low, I would greatly appreciate it.

Thank you in advance for your assistance!
